tests/test_kernels/lower_memory_free/src/bin/lower_memory_free.rs

```diff
@@ -0,0 +1,65 @@
+#![no_std] // don't link the Rust standard library
+#![no_main] // disable all Rust-level entry points
+
+use bootloader_api::{
+    config::Mapping, entry_point, info::MemoryRegionKind, BootInfo, BootloaderConfig,
+};
+use test_kernel_lower_memory_free::{exit_qemu, QemuExitCode};
+
+const LOWER_MEMORY_END_PAGE: u64 = 0x0010_0000;
+const WRITE_TEST_UNTIL: u64 = 0x4000_0000;
```

Comment:
```suggestion
const WRITE_TEST_END: u64 = 0x4000_0000;
```

I assume this exists because testing writing to the entire memory is too slow? If so please add a comment stating this.

Reply:
I just added this as some kind of defensive programming against slow memory access. But I just tested this and this is pretty much pointless.
1. We don't set a ram size for qemu and the default is 128MB (https://wiki.gentoo.org/wiki/QEMU/Options#RAM)
2. Even if I set this to the maximum on my computer (around 30GB) it only increases the total time of the test from 9.5s to 10.5s
So I will remove this.

Reply:
I also moved the write check into its own test, as it is not really related to the lower_memory_free test